src/v/cloud_topics/extent_meta.h

```diff
@@ -0,0 +1,55 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+#pragma once
+
+#include "cloud_topics/types.h"
+#include "model/fundamental.h"
+
+#include <fmt/core.h>
+
+namespace experimental::cloud_topics {
+
+// This struct contains extent information (location of the byte slice in the
+// object + the object uuid) and the kafka metadata (base offset and committed
+// offset).
+// Timestamps are not here because the timequery is handled in the control plane
```

Comment:
> control plane

I thought we had decided in slack to call this "metadata storage" or "metadata layer" instead of using the control plane and data plane. It's not really a control plane.